modules/exploits/windows/smb/ms17_010_eternalblue.rb

```diff
@@ -42,96 +45,1087 @@ def initialize(info = {})
-                'et_tle' => 0x420 # ETHREAD.ThreadListEntry offset
+                'os_patterns' => ['Windows 8.1']
+              }
+            ],
+            [
+              'Windows Server 2012',
+              {
+                'os_patterns' => ['Windows Server 2012']
+              }
+            ],
+            [
+              'Windows 10 Pro',
+              {
+                'os_patterns' => ['Windows Pro Build']
+              }
+            ],
+            [
+              'Windows 10 Enterprise Evaluation',
+              {
+                'os_patterns' => ['Windows 10 Enterprise Evaluation Build']
               }
             ]
           ],
         'DefaultTarget' => 0,
-        'DisclosureDate' => '2017-03-14',
         'Notes' =>
           {
             'AKA' => ['ETERNALBLUE']
-          }
+          },
+        'DisclosureDate' => '2017-03-14'
       )
     )
 
     register_options(
       [
+        Opt::RHOSTS,
         Opt::RPORT(445),
-        OptBool.new('VERIFY_TARGET', [true, 'Check if remote OS matches exploit Target.', true]),
-        OptBool.new('VERIFY_ARCH', [true, 'Check if remote architecture matches exploit Target.', true]),
         OptString.new('SMBUser', [false, '(Optional) The username to authenticate as', '']),
         OptString.new('SMBPass', [false, '(Optional) The password for the specified username', '']),
-        OptString.new('SMBDomain', [false, '(Optional) The Windows domain to use for authentication', '.'])
+        OptString.new('SMBDomain', [
+          false,
+          '(Optional) The Windows domain to use for authentication. Only affects Windows Server 2008 R2, Windows 7,' \
+            ' Windows Embedded Standard 7 target machines.',
+          ''
+        ]),
+        OptBool.new('VERIFY_TARGET', [
+          true,
+          'Check if remote OS matches exploit Target. Only affects Windows Server 2008 R2, Windows 7, Windows Embedded' \
+            ' Standard 7 target machines.',
+          true
+        ]),
+        OptBool.new('VERIFY_ARCH', [
+          true,
+          'Check if remote architecture matches exploit Target. Only affects Windows Server 2008 R2, Windows 7,' \
+            ' Windows Embedded Standard 7 target machines.',
+          true
+        ])
       ]
     )
     register_advanced_options(
       [
         OptString.new('ProcessName', [true, 'Process to inject payload into.', 'spoolsv.exe']),
-        OptInt.new('MaxExploitAttempts', [true, 'The number of times to retry the exploit.', 3]),
         OptInt.new('GroomAllocations', [true, 'Initial number of times to groom the kernel pool.', 12]),
-        OptInt.new('GroomDelta', [true, 'The amount to increase the groom count by per try.', 5])
+        OptInt.new('MaxExploitAttempts', [
+          true,
+          'The number of times to retry the exploit. Useful as EternalBlue can sometimes require multiple attempts to
+          get a successful execution.',
```

Comment:
Looks like there's an extra line here which breaks the `show advanced` word wrapping:
![image](https://user-images.githubusercontent.com/60357436/124595911-81702c80-de59-11eb-879b-230015f701e5.png)

```suggestion
          'The number of times to retry the exploit. Useful as EternalBlue can sometimes require multiple attempts to' \
          'get a successful execution.',
```